ledger-tool/src/main.rs

```diff
@@ -4185,55 +4189,140 @@ fn main() {
-                                    last
-                                );
-                                if all {
-                                    println!("Non-empty slots: {slots:?}");
+                            
+                            // Collect info about slot bounds
+                            let mut bounds = SlotBounds {
+                                slots: Some(SlotInfo {
+                                    total: slots.len(),
+                                    first: Some(*slots.first().unwrap()),
+                                    last: Some(*slots.last().unwrap()),
+                                    ..Default::default()
+                                }),
+                                ..Default::default()
+                            };
+
+                            // Consider also rooted slots, if present
+                            if let Ok(rooted) = blockstore.rooted_slot_iterator(0) {
+                                let mut first_rooted = 0;
+                                let mut last_rooted = 0;
+                                let mut total_rooted = 0;
+                                for (i, slot) in rooted.into_iter().enumerate() {
+                                    if i == 0 {
+                                        first_rooted = slot;
+                                    }
+                                    last_rooted = slot;
+                                    total_rooted += 1;
                                 }
-                            } else {
-                                println!("Ledger has data for slot {first:?}");
+
+                                let count_past_root = slots.iter()
+                                    .rev()
+                                    .take_while(|slot| *slot > &last_rooted)
+                                    .count();
+
+                                bounds.roots = Some(SlotInfo {
+                                    total: total_rooted,
+                                    first: Some(first_rooted),
+                                    last: Some(last_rooted),
+                                    num_after_last_root: Some(count_past_root)
+                                });
                             }
-                        }
-                        if let Ok(rooted) = blockstore.rooted_slot_iterator(0) {
-                            let mut first_rooted = 0;
-                            let mut last_rooted = 0;
-                            let mut total_rooted = 0;
-                            for (i, slot) in rooted.into_iter().enumerate() {
-                                if i == 0 {
-                                    first_rooted = slot;
-                                }
-                                last_rooted = slot;
-                                total_rooted += 1;
+                            bounds
+                        };
+
+                        // if slots.is_empty() {
```

Comment:
Just incase you didn't know, you can do block comments like this:
```rust
/*
all
of
these
lines
will
be
commented
*/
```
Less tedious than commenting out each individual line

Reply:
Oh sure, i was just using VS code shortcuts for commenting (ctrl+k+c) :)

Reply:
Ha nice, figured as much